keep/api/routes/incidents.py

```diff
@@ -473,6 +478,81 @@ def delete_alerts_from_incident(
     return Response(status_code=202)
 
 
+@router.post(
+    "/event/{provider_type}",
+    description="Receive an alert event from a provider",
+    status_code=202,
+)
+async def receive_event(
+    provider_type: str,
+    bg_tasks: BackgroundTasks,
+    request: Request,
+    provider_id: str | None = None,
+    event=Depends(extract_generic_body),
+    authenticated_entity: AuthenticatedEntity = Depends(
+        IdentityManagerFactory.get_auth_verifier(["write:incident"])
+    ),
+    pusher_client: Pusher = Depends(get_pusher_client),
+) -> dict[str, str]:
+    trace_id = request.state.trace_id
+
+    provider_class = None
+    try:
+        provider_class = ProvidersFactory.get_provider_class(provider_type)
+    except ModuleNotFoundError:
+        raise HTTPException(
+            status_code=400, detail=f"Provider {provider_type} not found"
+        )
+    if not provider_class:
+        raise HTTPException(
+            status_code=400, detail=f"Provider {provider_type} not found"
+        )
+
+    # Parse the raw body
+    event = provider_class.parse_event_raw_body(event)
+
+    if REDIS:
+        redis: ArqRedis = await get_pool()
+        job = await redis.enqueue_job(
+            "async_process_event",
+            authenticated_entity.tenant_id,
+            provider_type,
+            provider_id,
+            None,
+            authenticated_entity.api_key_name,
+            trace_id,
+            event,
+            True,
+            None,
+            "incident",
+            _queue_name=KEEP_ARQ_QUEUE_BASIC,
+        )
+        logger.info(
+            "Enqueued job",
+            extra={
+                "job_id": job.job_id,
+                "tenant_id": authenticated_entity.tenant_id,
+                "queue": KEEP_ARQ_QUEUE_BASIC,
+            },
+        )
+    else:
+        bg_tasks.add_task(
+            process_event,
+            {},
+            authenticated_entity.tenant_id,
+            provider_type,
+            provider_id,
+            None,
+            authenticated_entity.api_key_name,
+            trace_id,
+            event,
+            True,
+            None,
+            "incident"
+        )
+    return Response(status_code=202)
```

Comment:
What do you think if we create a new task for this and not overload the "process_event_task"? We want to refactor it anyhow.. cc: @shahargl @VladimirFilonov

Reply:
On it.